app/controllers/declaration/PackageInformationController.scala

```diff
@@ -60,68 +60,54 @@ class PackageInformationController @Inject()(
           val packagings = data.getOrElse(Seq.empty)
 
           actionTypeOpt match {
-            case Some(Add)             => addItem(itemId, packagings)
-            case Some(Remove(ids))     => remove(itemId, packagings, boundForm, ids)
-            case Some(SaveAndContinue) => continue(itemId, packagings)
+            case Some(Add)             => handleAdd(itemId, boundForm, packagings)
```

Comment:
I better like first naming convention - having everywhere handle is little meaningless

Reply:
Maybe it's because I'm old-school - I think of them as event handlers.   But in fairness, I copied most of the changes in this controller from the AdditionaltemsController.

As for your other comments, I'll check them next week.  I just edited the tests to get them to compile after I made the form elements non-optional.